administrator/modules/mod_quickicon/Helper/QuickIconHelper.php

```diff
@@ -53,194 +55,235 @@ public static function &getButtons(Registry $params, CMSApplication $application
 			$application = Factory::getApplication();
 		}
 
-		$context = $params->get('context', 'mod_quickicon');
-
 		$key = (string) $params;
 
 		if (!isset(self::$buttons[$key]))
 		{
-			// Load mod_quickicon language file in case this method is called before rendering the module
-			$application->getLanguage()->load('mod_quickicon');
+			$context = (string) $params->get('context', 'mod_quickicon');
 
-			// Update Panel, icons come from plugins quickicons
-			if ($params->get('show_jupdate'))
-			{
-				PluginHelper::importPlugin('quickicon', 'joomlaupdate');
-			}
+			self::$buttons[$key] = [];
 
-			if ($params->get('show_eupdate'))
-			{
-				PluginHelper::importPlugin('quickicon', 'extensionupdate');
-			}
+			// @TODO: Perhaps we should use something with EventAware
+			$container = (new Container)
```

Comment:
What are you doing here?  You're creating a new DI container and creating a new event dispatcher.  For what purpose?  There is exactly one event dispatcher in a Joomla application.  I assume this is for your "scope" of distinguishing different contexts or "core" versus "third party", these are the types of checks that belong in the plugin, not bypassing the global event dispatcher.

The entire redesign of this module is beyond wrong.  There should not be a distinction between core extensions and third party extensions.  What makes something core versus third party?  If someone adds a quickicon for weblinks, is that core or third party?  Why should a module control what plugins are or are not loaded, this is the type of behavior that needs to be erased from Joomla core code (expanded on below).

There should never be any code added in core for specific plugins.  This means that the capability is not something that should be a plugin in the first place.  This has been a problem in core for a long time and requires effort to fix, there should not be new code introduced using this anti-pattern.  Therefore, the options in this module to show or hide the result from a plugin should be removed, or the plugins should be removed and their capabilities inlined into this module.

And as @brianteeman has said about a thousand times, this is the type of change that does not belong in this repository but should have been made in the main repository.  There has been a massive lack of code review in the quickicon module changes and the result is more people are going to have to invest more time into fixing things, assuming they can discover the problems.

Reply:
I am sure I only said it a hundred times not a thousand ;)